### ISSUE TITLE
SplPriorityQueue::getExtractFlags() is flagged as undocumented

### ISSUE THREAD

**Issue by ollieread** [php/doc-en#2199]
The [`SplPriorityQueue::getExtractFlags()`](https://www.php.net/manual/en/splpriorityqueue.getextractflags.php) method is marked as undocumented, but it has no less documentation than any of the other methods. Should the flag be removed, or should the others have it added?

From what I can tell, the only thing missing is examples, which they all miss. Is there a definitely somewhere as to what constitutes "undocumented" or "underdocumented"? I've compiled a list (markdown and json) [here](https://gist.github.com/ollieread/c997de434ea01a960f6ca2e363aa3414) of all things currently flagged as such.

**Comment by cmb69** [comment]
No, there are now hard rules for when this "flag" should be there or not. I think, a function/method documentation should be self-explaining, or link to relevant info. Examples are not strictly necessary.

In case of `SplPriorityQueue::getExtractFlags()` the supported flags should be listed like for `SplPriorityQueue::setExtractFlags()`, or there should at least be a link to that listing. Or probably even better, the flag descriptions should be moved to the [class page](https://www.php.net/manual/en/class.splpriorityqueue.php), and the methods should link to that.

Oh, and we should add see-also sections for both methods.

**Comment by ollieread** [comment]
> No, there are now hard rules for when this "flag" should be there or not. I think, a function/method documentation should be self-explaining, or link to relevant info. Examples are not strictly necessary.
> 
> In case of `SplPriorityQueue::getExtractFlags()` the supported flags should be listed like for `SplPriorityQueue::setExtractFlags()`, or there should at least be a link to that listing. Or probably even better, the flag descriptions should be moved to the [class page](https://www.php.net/manual/en/class.splpriorityqueue.php), and the methods should link to that.
> 
> Oh, and we should add see-also sections for both methods.

So if I'm understanding correctly, the following is a sensible list of tasks for this?

- [ ] Add `EXTR_*` descriptions to the `SplPriorityQueue` page.
- [ ] Add "See Also" sections so that:
    - [ ] `getExtractFlags()` links to `setExtractFlags()`, `extract()`, `current()`, `top()` and each constant.
    - [ ] `setExtractFlags()` links to `getExtractFlags()`, `extract()`, `current()`, `top()` and each constant.
    - [ ] `extract()` links to `getExtractFlags()`, `setExtractFlags()`, `current()`, `top()` and each constant.
    - [ ] `current()` links to `getExtractFlags()`, `extract()`, `setExtractFlags()`, `top()` and each constant.
    - [ ] `top()` links to `getExtractFlags()`, `extract()`, `current()`, `setExtractFlags()` and each constant.
- [ ] Add "Examples" to `extract()`, `current()` and `top()` to show:
    - [ ] Using `EXTR_DATA`
    - [ ] Using `EXTR_PRIORITY`
    - [ ] Using `EXTR_BOTH`

To be honest, we could probably do with some more example for this whole class, as the following doesn't really explain much.

> The SplPriorityQueue class provides the main functionalities of a prioritized queue, implemented using a max heap. 

I appreciate that's what it does, but there's not going to mean much to many people.
